packages/myst-cli/src/frontmatter.ts

```diff
@@ -83,7 +83,7 @@ export function prepareToWrite(frontmatter: { license?: Licenses }) {
 
 export async function getRawFrontmatterFromFile(session: ISession, file: string) {
   const cache = castSession(session);
-  await loadFile(session, file);
+  if (!cache.$mdast[file]) await loadFile(session, file);
```

Comment:
This tiny change is all that was required to fix the bug addressed in the top PR comment 😆